### PR TITLE
added htmx to delete questions from a survey

### PR DESCRIPTION
**Pull request by DavidHaighton**

### CODE REVIEW

File: src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java [hunk 1]
```diff
@@ -55,6 +55,16 @@ public String deleteQuestion(@RequestParam long questionID, @RequestParam long s
         return "redirect:/survey/"+surveyID +"/edit";
     }
 
+    @DeleteMapping(value = "question/delete/{questionID}")
+    @ResponseBody
+    public String deleteSurveyHtmx(@PathVariable long questionID){
```
Comment (Shashaank5) [inline src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java:60]: What exactly does the @responsebody string do given that you're returning a string

File: src/main/resources/templates/newSurvey.html [hunk 2]
```diff
@@ -37,11 +39,9 @@ <h2 th:text="'Survey Name: '+${survey.getName()}">Survey Name</h2>
             <td th:text="${question.getId()}"></td>
             <td><a th:href="@{/question/{q}(q=${question.getId()})}" th:text="${question.getPrompt()}"></a></td>
             <td>
-                <form method="post" th:action="@{/question/delete}">
```
Comment (Shashaank5) [inline src/main/resources/templates/newSurvey.html:40]: Nice job, it's good we're getting rid of this janky code

**Review by Shashaank5 — APPROVED** [review]

**Review by Shashaank5 — COMMENTED** [review]
Good job on the changes but you're missing some tests like integration, add that then you're good to go

File: src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java (hunk 1 above)
Comment (DavidHaighton) [reply src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java:60]: it makes it so we don't return the path to a html view.

File: src/main/resources/templates/newSurvey.html (hunk 2 above)
Comment (DavidHaighton) [reply src/main/resources/templates/newSurvey.html:40]: htmx is pretty nice

**Comment by DavidHaighton** [comment]
> Good job on the changes but you're missing some tests like integration, add that then you're good to go

Nice catch!

**Review by Shashaank5 — APPROVED** [review]
Looks good

**Pull request merged**